pinot-common/src/main/java/org/apache/pinot/sql/parsers/rewriter/NonAggregationGroupByToDistinctQueryRewriter.java

```diff
@@ -57,6 +57,13 @@ public PinotQuery rewrite(PinotQuery pinotQuery) {
       Set<String> selectIdentifiers = CalciteSqlParser.extractIdentifiers(pinotQuery.getSelectList(), true);
       Set<String> groupByIdentifiers = CalciteSqlParser.extractIdentifiers(pinotQuery.getGroupByList(), true);
       if (groupByIdentifiers.containsAll(selectIdentifiers)) {
+        // TODO: need to patch DistinctDataTableReducer to remove these excessive columns
```

Comment:
See #9605 which fixes the wrong behavior in V1 and should cover V2 correctly

Reply:
made changes accordingly, let's see the result

Reply:
Actually with the v2 engine extra logic. No need for any change in v1 rewrite.

In fact after the PR v2 engine guarantee the select/groupby list are the same